File: decision_maker/dice.py
import random
from typing import NamedTuple


class Dice(NamedTuple):
    probabilities: list[float] = [0.1, 0.2, 0.3, 0.2, 0.1, 0.1]

    def roll(self) -> int:
        return random.choices(range(1, 7), weights=self.probabilities)[0]

File: dice/dice.go
package dice

import (
    "math/rand"
)

type Dice struct {
    probabilities []float64
}
func New() Dice {
    return Dice{probabilities: []float64{0.1, 0.2, 0.3, 0.2, 0.1, 0.1}}
}
func (self Dice) Roll() int {
    var sum float64 = 0.0
    var r float64 = rand.Float64()
    for i := 0; i < len(self.probabilities); i++ {
        sum += self.probabilities[i]
        if r < sum {
            return i + 1
        }
    }
    return len(self.probabilities)  
}
